diff -Naur glib-2.32.4.orig/glib/gutils.h glib-2.32.4/glib/gutils.h
--- glib-2.32.4.orig/glib/gutils.h	2012-07-14 12:17:32.000000000 -0700
+++ glib-2.32.4/glib/gutils.h	2013-04-18 13:12:44.871762301 -0700
@@ -216,7 +216,6 @@
 GLIB_DEPRECATED_FOR(g_format_size)
 gchar *g_format_size_for_display (goffset size);
 
-#ifndef G_DISABLE_DEPRECATED
 /**
  * GVoidFunc:
  *
@@ -235,7 +234,6 @@
  * (if there is any in the implementation) and doesn't encounter
  * missing include files.
  */
-GLIB_DEPRECATED
 void	g_atexit		(GVoidFunc    func);
 
 #ifdef G_OS_WIN32
@@ -251,8 +249,6 @@
 #define g_atexit(func) atexit(func)
 #endif
 
-#endif  /* G_DISABLE_DEPRECATED */
-
 #ifndef __GTK_DOC_IGNORE__
 #ifdef G_OS_WIN32
 #define g_find_program_in_path g_find_program_in_path_utf8
